File: datasets/JPlag-Java/to-kotlin-intellij/TryWithResource.kt
package de.jplag.java

import java.io.File
import java.io.FileNotFoundException
import java.util.*

class TryWithResource {
    fun load(path: String?) {
        var other: Scanner? = null // This is just here to keep the tokens similar.
        try {
            Scanner(File(path)).also { other = it }.use { scanner ->  // same for = other =
                while (scanner.hasNext()) {
                    println(scanner.nextLine())
                }
            }
        } catch (exception: FileNotFoundException) {
            exception.printStackTrace()
        } finally {
            if (other != null) { // This as well...
                other!!.close() // This as well...
            }
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            TryWithResource().load("DoesNotExist.txt")
        }
    }
}

File: datasets/JPlag-Java/to-python-convertai/TryWithResource.py
import os

def load(path):
    try:
        with open(path, 'r') as scanner:
            for line in scanner:
                print(line.strip())
    except FileNotFoundError:
        print(f"Error: {path} not found.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    load("DoesNotExist.txt")

